blockstack_client/data.py

```diff
@@ -278,39 +268,37 @@ def list_update_history( name, current_block=None, proxy=None ):
     Return None on error
     """
 
-    if proxy is None:
-        proxy = get_default_proxy()
+    proxy = get_default_proxy() if proxy is None else proxy
 
     if current_block is None:
-        try:
-            info = proxy.getinfo()
-            if info.has_key('last_block_processed'):
-                current_block = int(info['last_block_processed']) + 1
-            elif info.has_key('last_block'):
-                current_block = int(info['last_block']) + 1
-            else:
-                raise Exception("Invalid getinfo reply")
-        except Exception, e:
-            log.error("Invalid getinfo reply")
+        info = proxy.getinfo()
```

Comment:
Part of the original `try/except` block's job earlier was to catch invalid data returned by `proxy.getinfo()`.  I'm working on a patch to `proxy.py` that raises an exception of the server-replied data doesn't match our schema, but the `try/except` block should still be here to cover that case.